webui/react/src/components/HpSelect.tsx

```diff
@@ -25,10 +25,11 @@ const HpSelect: React.FC<Props> = ({ fullHParams, onChange, value, ...props }: P
         onChange([], option);
         if (document.activeElement) (document.activeElement as HTMLElement).blur();
       } else {
-        const newValue = clone(values);
+        const newValue = structuredClone(values);
         const selectedValue = isObject(selected) ? (selected as LabeledValue).value : selected;
 
-        if (!newValue.includes(selectedValue)) newValue.push(selectedValue);
+        if (!!selectedValue && !Array.isArray(selectedValue) && !newValue.includes(selectedValue))
```

Comment:
maybe explicitly filter out undefined from selectedValue, otherwise `0` and `""` will also be caught out with this:
```suggestion
        if (selectedValue !== undefined && !Array.isArray(selectedValue) && !newValue.includes(selectedValue))
```